cockpit/IPC/src/views/Configurator.tsx

```diff
@@ -50,8 +50,26 @@ const Configurator: React.FC = () => {
         ) && !name.includes('ipc_ruler'),
       );
 
+      const sortedNames = filteredNames.sort((a, b) => {
+        const regex = /(\D+)(\d+)$/;
+        const matchA = a.match(regex);
+        const matchB = b.match(regex);
+
+        if (matchA && matchB) {
```

Comment:
Should we not check the length here just to be sure?

Reply:
> Should we not check the length here just to be sure?

`matchA` and `matchB` will either be null (if no match is found) or an array where:

- matchA[0] or matchB[0] is the full string that matches the regex.
- matchA[1] or matchB[1] is the first captured group (non-numeric part).
- matchA[2] or matchB[2] is the second captured group (numeric part).

The condition if (matchA && matchB) is checking if both matchA and matchB are not null, which means both strings matched the regex pattern. If this condition is true, then it is guaranteed that both matchA and matchB are arrays with at least three elements (the entire match, the non-numeric part, and the numeric part). Therefore, it's safe to access matchA[1], matchA[2], matchB[1], and matchB[2] without an additional length check.